### DISCUSSION TITLE
How to enable Mutex/Rwlock guard span?

### DISCUSSION THREAD

**Discussion by MrCroxx** [tokio-rs/tokio#4701] [Q&A] [answered]
When I am diagnosing deadlock issue in my application, I'd to print lock guard span logs.

I found there's `resource_span` field in the guards of `Mutex` and `RwLock` with `#[cfg(all(tokio_unstable, feature = "tracing"))]` attribute. And I built my application with `RUSTFLAGS="--cfg tokio_unstable"` and enabled `tracing` feature of tokio. Then run my appliaction with `RUST_LOG=trace`. But I still got no logs of tokio mutex/rwlock.

Did I miss something? Could tokio user enable mutex/rwlock guard span?

(Plus, is there any better way to detect deadlock with tokio? `parking_lot` has `deadlock` feature to detect deadlock. Will tokio has the feature in future?

**Comment by MrCroxx** [comment]
I retried in a brand-new project and that works. Seems I didn't configure my subscriber right in the origin project.

**Comment by Darksonn** [comment, marked as answer]
Unfortunately, detecting deadlocks is not as easy with async locks as it is for something like `parking_lot`. There could be things like a `tokio::select!` that the lock doesn't know about, but which can resolve the deadlock by cancelling a call to `lock`.
